### ISSUE TITLE
Feature: warn about single quotes in variable defaults

### ISSUE THREAD

**Issue by fedgiac** [koalaman/shellcheck#2150]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

Double quotes and single quotes behave differently than usual in a default value for a bash variable:
```
$ echo "Single: ${UNSET-''}, double: ${UNSET-""}, no quotes: ${UNSET-}"
Single: '', double: , no quotes: 
```

This could lead to issues in scripts like this:
```
#!/bin/bash
set -o nounset

if [ -z "${UNSET-''}" ]; then
        echo "Variable is unset or empty"
fi
```
Since `''` isn't empty, the if condition is unexpectedly not triggered if the variable is unset.

My Bash version is `5.0.17(1)-release`.

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

A warning that single quotes cannot be used to enclose strings in default values.
